internal/usecases/user/usecase.go

```diff
@@ -34,37 +34,22 @@ type Usecases struct {
 }
 
 // Create Add new user
-func (uc *Usecases) Create(ctx context.Context, nu entity.NewUser) (string, error) {
-	hash, err := bcrypt.GenerateFromPassword([]byte(nu.Password), bcrypt.DefaultCost)
+func (uc *Usecases) Create(ctx context.Context, u *entity.User) error {
+	hash, err := bcrypt.GenerateFromPassword([]byte(u.PasswordHash), bcrypt.DefaultCost)
 	if err != nil {
 		return "", fmt.Errorf("generate password hash:%w", err)
 	}
 
-	id := uuid.New().String()
+	u.ID = uuid.New().String()
+	u.PasswordHash = string(hash)
+	u.PermissionRole = role.Viewer
 
-	u := entity.User{
-		ID:             id,
-		Name:           nu.Name,
-		Email:          nu.Email,
-		PermissionRole: role.Viewer,
-		PasswordHash:   hash,
-	}
-	return id, uc.repo.Create(ctx, u)
+	return uc.repo.Create(ctx, u)
 }
 
 // Update updates a user
-func (uc *Usecases) Update(ctx context.Context, nu entity.NewUser) (entity.User, error) {
-	hash, err := bcrypt.GenerateFromPassword([]byte(nu.Password), bcrypt.DefaultCost)
-	if err != nil {
-		return entity.User{}, fmt.Errorf("generate password hash:%w", err)
-	}
-	u := entity.User{
-		ID:           nu.ID,
-		Name:         nu.Name,
-		Email:        nu.Email,
-		PasswordHash: hash,
-	}
-	return u, uc.repo.Update(ctx, &u)
+func (uc *Usecases) Update(ctx context.Context, u *entity.User) error {
+	return uc.repo.Update(ctx, u)
```

Comment:
probably, we should add entities for each level  in future

Reply:
hm ?

Reply:
like request entity, db entity... etc. 
not now

Reply:
no